Doc/#276 ParquetDataNode documentation (#298)
Avaiga/taipy-doc: docs/manuals/core/config/data-node-config.md

    To configure a JSON data node, it is equivalent to use the method `Config.configure_json_data_node()^` or
    the method `Config.configure_data_node()^` with parameter `storage_type="json"`.

# Parquet

!!! Info

>     The ParquetDataNode implementation in Taipy wraps _pandas_ (itself a wrapper of the _pyarrow_ and _fastparquet_ engines) through the [`pandas.read_parquet`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_parquet.html) and [`pandas.DataFrame.to_parquet`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_parquet.html) methods for reading and writing Parquet data respectively.

Reviewer: ```suggestion
    Taipy ParquetDataNode wraps [`pandas.read_parquet`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_parquet.html) and [`pandas.DataFrame.to_parquet`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_parquet.html) methods for reading and writing Parquet data, respectively.
```
Just shorten the sentence a bit.

However, I don't think this should be here at the start of the section.

The user only needs to know about the `pandas.DataFrame.read_parquet` when using the `read_kwargs` or the `read_with_kwargs()` method. This should be described somewhere else probably.

Reviewer 2: > However, I don't think this should be here at the start of the section.
> 
> The user only needs to know about the pandas.DataFrame.read_parquet when using the read_kwargs > or the read_with_kwargs() method. This should be described somewhere else probably.

Yes I agree

A `ParquetDataNode^` data node is a specific data node used to model [Parquet](https://parquet.apache.org/) file data. To add a new _Parquet_ data node
configuration, the `Config.configure_parquet_data_node()^` method can be used. In addition to the generic parameters

> described in the previous section [Data node configuration](data-node-config.md), the following parameters can be provided:

Reviewer: ```suggestion
described in [Data node configuration](data-node-config.md) section, the following parameters can be provided:
```
This is a systematic error, probably we need to fix it all.

> -   _**default_path**_ is a mandatory parameter and represents the default Parquet file path used by Taipy to read and write

Reviewer: The path can be a file or a folder. We should clarify that.

    the data.

-   _**engine**_ represents the Parquet library to use.<br/>

>     Possible values are _"fastparquet"_ or _"pyarrow"_. The default value is _"pyarrow"_.

Reviewer: We haven't got a small notice where we clarify that if the user wants to use "fastparquet", he should install taipy with `pip install taipy[fastparquet]` yet.
What do you think?

-   _**compression**_ is the name of the compression to use.<br/>
    Possible values are _"snappy"_, _"gzip"_, _"brotli"_ and `None`. The default value is _"snappy"_. Use None for no compression.

> -   _**read_kwargs**_ is a dictionary of additional parameters passed to the _pandas.read_parquet_ method.

Reviewer: ```suggestion
-   _**read_kwargs**_ is a dictionary of additional parameters passed to the [`pandas.read_parquet`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_parquet.html)  method.
```

> -   _**write_kwargs**_ is a dictionary of additional parameters passed to the _pandas.DataFrame.write_parquet_ method.<br/>

Reviewer: ```suggestion
-   _**write_kwargs**_ is a dictionary of additional parameters passed to the [`pandas.DataFrame.to_parquet`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_parquet.html) method.<br/>
```

>     The parameters in _"read_kwargs"_ and _"write_kwargs"_ have a **higher precedence** than the top-level parameters (**engine** and **compression**) which are also passed to Pandas.

Reviewer: ```suggestion
    The parameters in _"read_kwargs"_ and _"write_kwargs"_ have a **higher precedence** than the top-level parameters (**engine** and **compression**) which are also passed to Pandas. Passing `read_kwargs={"engine": "fastparquet", "compression": "gzip"} will override the **engine** and **compression** properties of the data node.
```
Now think about this, I don't think we want the user to override **engine** and **compression** properties though ...

> 

Reviewer: Probably the information about `pandas.DataFrame.read_parquet` method should be here so that the user can look up for possible `read_` or `write_kwargs`.

    !!! Tip

>         Some parameters (such as _"columns"_, used for reading a subset of columns from the data) only make sense to be passed to the Parquet engine at runtime instead of in the configuration. The `ParquetDataNode.read_with_kwargs^` and `ParquetDataNode.write_with_kwargs^` methods provide an alternative for specifying keyword arguments at runtime. See examples of using this method at the [Data Node Management page](../entities/data-node-mgt.md#parquet).

Reviewer: ```suggestion
        The `ParquetDataNode.read_with_kwargs^` and `ParquetDataNode.write_with_kwargs^` methods provide an alternative for specifying keyword arguments at runtime. See examples of these methods at the [Data Node Management page](../entities/data-node-mgt.md#parquet).
```
The tips is great.

Btw, since you've provide examples on data-node-mgt page, the first sentence can be moved to that page, or removed since it's a pandas functionality. What do you think?

    -   By default, _exposed_type_ is "pandas", and the data node will read the Parquet file as a `pandas.DataFrame`.
    -   If the _exposed_type_ value provided is "modin", the data node will read the Parquet file as a `modin.pandas.DataFrame`.
    -   If the _exposed_type_ value provided is "numpy", the data node will read the Parquet file as a numpy array.
    -   If the provided _exposed_type_ value is a Callable, the data node will create a list of objects as returned by the Callable. Each object will represent a row in the Parquet file. The Parquet file is read as a `pandas.DataFrame`, and each row of the DataFrame is passed to the Callable as keyword arguments where the key is the column name and the value is the corresponding value for that row. 

```python linenums="1"
from taipy import Config

temp_cfg = Config.configure_parquet_data_node(
    id="historical_temperature",
    default_path="path/hist_temp.parquet",
    engine="pyarrow", # default
    compression="snappy", # default

>     exposed_type="pandas") # default

Reviewer: ```suggestion
```
If set as default, these are not necessary I guess.

You have another config example that specify how to use these parameters.
